src/connector/codec/src/decoder/mod.rs

```diff
@@ -44,33 +44,40 @@ pub enum AccessError {
 pub type AccessResult<T = Datum> = std::result::Result<T, AccessError>;
 
 /// Access to a field in the data structure.
-///
-/// Only one of these two methods should be implemented. See documentation for more details.
 pub trait Access {
     /// Accesses `path` in the data structure (*parsed* Avro/JSON/Protobuf data),
     /// and then converts it to RisingWave `Datum`.
     /// `type_expected` might or might not be used during the conversion depending on the implementation.
     ///
+    /// # Path
+    ///
     /// We usually expect the data is a record (struct), and `path` represents field path.
     /// The data (or part of the data) represents the whole row (`Vec<Datum>`),
     /// and we use different `path` to access one column at a time.
     ///
     /// e.g., for Avro, we access `["col_name"]`; for Debezium Avro, we access `["before", "col_name"]`.
-    fn access(&self, path: &[&str], type_expected: &DataType) -> AccessResult<Datum> {
-        self.access_cow(path, type_expected)
-            .map(ToOwnedDatum::to_owned_datum)
-    }
-
     /// Similar to `access`, but may return a borrowed [`DatumCow::Borrowed`] to avoid unnecessary allocation.
     /// If not overridden, it will call forward to `access` and always wrap the result in [`DatumCow::Owned`].
     ///
     /// This should be preferred over `access` for both callers and implementors.
```

Comment:
The doc comment needs to be fixed. It's mixed together with above.

Reply:
It is likely due to resolving merge conflicts. Will fix.